### ISSUE TITLE
Cannot fake type T with ILogger<T> in the contructor

### ISSUE THREAD

**Issue by bluecitylights** [Serg046/AutoFake#142]
This test below fails, because AutoFake cannot find the constructor
```
public class TestClass
        {
            private readonly ILogger<TestClass> _logger;
            public TestClass(ILogger<TestClass> logger)
            {
                _logger = logger;
            }

            public void DoSometing()
            {

            }
        }

        [TestMethod]
        public void TestLogger()
        {
            var logger = Substitute.For<ILogger<TestClass>>();
            var fake = new Fake<TestClass>(logger);
            var sut = fake.Rewrite(f => f.DoSometing());
            sut.Execute();
        }
```

```
PlanOutcomePredicting.Test.PlanStatisticsServiceTests.TestLogger:
    Outcome: Failed
    Error Message:
    Test method PlanOutcomePredicting.Test.PlanStatisticsServiceTests.TestLogger threw exception: 
AutoFake.Exceptions.InitializationException: Constructor is not found
    Stack Trace:
        at AutoFake.TypeInfo.CreateInstance(Type type)
   at AutoFake.TypeInfo.CreateFakeObject(MockCollection mocks)
   at AutoFake.Fake.GetFakeObject()
   at AutoFake.ExecutorImpl.Execute()
   at AutoFake.Executor.Execute()
   at AutoFake.Setup.Configurations.ActionMockConfiguration`1.Execute()
```

**Comment by Serg046** [comment]
Fixed and released as v0.4.3.5. Unfortunately there are still quite much stuff to cover...

**Comment by bluecitylights** [comment]
seems to work now. thanks for the quick response.
